packages/visualizations/addon/serializers/table.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2018, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import VisualizationSerializer from 'navi-visualizations/serializers/visualization';
+import { parseMetricName } from 'navi-data/utils/metric';
+import { get } from '@ember/object';
+import { isPresent } from '@ember/utils';
+
+export default VisualizationSerializer.extend({
+  /**
+   * Normalizes payload so that it can be applied to models correctly
+   * @param type - class type as a DS model
+   * @param visualization - json parsed object
+   * @return {Object} normalized payload
+   */
+  normalize(type, visualization) {
+    if(visualization) {
+      let columns = get(visualization, 'metadata.columns');
+
+      if(isPresent(columns)) {
+        columns.map(column => {
```

Comment:
This is more of a `forEach` than a `map`. Use `map` to return a new array and then assign it to `metadata.columns`

Instead of a switch can we use an object:
```js
const TRANSFORMS = {
   metric:    (column) => column,
   threshold: (column) => assign(column, { field: parseMetricName(column.field) }),
   dimension: (column) => assign(column, { field: { dimension: column.field } })
   dateTime:  (column) => assign(column, { field: { dateTime: column.field } })
};


...


normalize(type, visualization) {
  if(visualization) {
    let columns = get(visualization, 'metadata.columns').map(column => {
      if(typeof get(column, 'field') === 'string') {
        let type = get(column, 'type');
        return TRANSFORMS[type](column);
      }
      return column;
    });
    assign(visualization.metadata, { columns });
  }
  return this._super(type, visualization);
}

```